airflow/utils/logging.py

```diff
@@ -211,3 +212,30 @@ def parse_gcs_url(self, gsurl):
             bucket = parsed_url.netloc
             blob = parsed_url.path.strip('/')
             return (bucket, blob)
+
+
+def get_log_filename(**kwargs):
+    """
+    Return full log filename. For example:
+        dag_id/task_id/execution_date/try_number.log
+    """
+    relative_dir = get_log_directory(**kwargs)
+    try_number = kwargs.get('try_number')
+    if try_number is None:
+        raise AirflowException("Try number is required to get log filename")
+
+    # Note try_number index starts from 1.
```

Comment:
Move the +1 to the part where you are setting try number initially.

Also do we need to increment try number below? It seems like it would be simpler to just maintain the same try number indexing throughout the code so we don't need to increment it like this anywhere.